core/vcpu.c

```diff
@@ -2581,30 +2581,27 @@ static void handle_cpuid_virtual(struct vcpu_t *vcpu, uint32_t a, uint32_t c)
             FEATURE(PSE)        |
             FEATURE(HTT);
 
-    static uint32_t cpu_features_2 =
+    static uint32_t cpuid_1_features_ecx =
             FEATURE(SSE3)       |
             FEATURE(SSSE3)      |
             FEATURE(SSE41)      |
             FEATURE(SSE42)      |
             FEATURE(CMPXCHG16B) |
             FEATURE(MOVBE)      |
+            FEATURE(AESNI)      |
+            FEATURE(PCLMULQDQ)  |
             FEATURE(POPCNT);
 
-    uint32_t cpu_features_ext =
+    static uint32_t cpuid_8000_0001_features_edx =
             FEATURE(NX)         |
             FEATURE(SYSCALL)    |
+            FEATURE(RDTSCP)     |
+            FEATURE(PDPE1GB)    |
```

Comment:
Seems like current implementation doesn't add PDPE1GB support yet. Is there any specific case that we can verify the correctness of this one?

Reply:
So far I understand, haxm code supports 4-level paging, so I added this feature bit. I have no tests for this feature.

Reply:
OK then, please move the adding of PDPE1GB, LAHF and PREFETCHW to a separate patch, so that we can bi-sect in case we found any issues during the full test, which generally took long time.